elasticsearch.go

```diff
@@ -175,99 +171,81 @@ func (es *Elasticsearch) CreateUser(ctx context.Context, statements dbplugin.Sta
-		errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to create user name %s: {{err}}", username), err))
+	if err := client.DeleteUser(ctx, req.Username); err != nil {
+		errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to delete user name %s: {{err}}", req.Username), err))
 	}
-	return errs
+	return newdbplugin.DeleteUserResponse{}, errs
 }
 
-// RotateRootCredentials doesn't require any statements from the user because it's not configurable in any
+// UpdateUser doesn't require any statements from the user because it's not configurable in any
 // way. We simply generate a new password and hit a pre-defined Elasticsearch REST API to rotate them.
-func (es *Elasticsearch) RotateRootCredentials(ctx context.Context, _ []string) (map[string]interface{}, error) {
-	newPassword, err := es.credentialProducer.GeneratePassword()
-	if err != nil {
-		return nil, errwrap.Wrapf("unable to generate root password: {{err}}", err)
-	}
+func (es *Elasticsearch) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
 
 	// Don't let anyone read or write the config while we're in the process of rotating the password.
 	es.mux.Lock()
 	defer es.mux.Unlock()
 
 	client, err := buildClient(es.config)
 	if err != nil {
-		return nil, errwrap.Wrapf("unable to get client: {{err}}", err)
+		return newdbplugin.UpdateUserResponse{}, fmt.Errorf("unable to get client: %w", err)
 	}
 
-	if err := client.ChangePassword(ctx, es.config["username"].(string), newPassword); err != nil {
-		return nil, errwrap.Wrapf("unable to change password: {{}}", err)
+	if req.Password != nil {
+		if err := client.ChangePassword(ctx, req.Username, req.Password.NewPassword); err != nil {
+			return newdbplugin.UpdateUserResponse{}, fmt.Errorf("unable to change password: %w", err)
+		}
+
 	}
 
-	es.config["password"] = newPassword
-	return es.config, nil
+	return newdbplugin.UpdateUserResponse{}, nil
 }
 
+// Close for Elasticsearch is a NOOP, nothing to close
 func (es *Elasticsearch) Close() error {
-	// NOOP, nothing to close.
 	return nil
 }
 
-// DEPRECATED, included for backward-compatibility until removal
-func (es *Elasticsearch) Initialize(ctx context.Context, config map[string]interface{}, verifyConnection bool) error {
-	_, err := es.Init(ctx, config, verifyConnection)
-	return err
-}
-
-func newCreationStatement(statements dbplugin.Statements) (*creationStatement, error) {
-	if len(statements.Creation) == 0 {
+func newCreationStatement(statements newdbplugin.Statements) (*creationStatement, error) {
+	if len(statements.Commands) == 0 {
 		return nil, dbutil.ErrEmptyCreationStatement
 	}
 	stmt := &creationStatement{}
-	if err := json.Unmarshal([]byte(statements.Creation[0]), stmt); err != nil {
-		return nil, errwrap.Wrapf(fmt.Sprintf("unable to unmarshal %s: {{err}}", []byte(statements.Creation[0])), err)
+	if err := json.Unmarshal([]byte(statements.Commands[0]), stmt); err != nil {
```

Comment:
If I'm reading this correctly, this function will ignore any commands that aren't in the first index. If so, can you put an error if the length of the commands is >1 so the user doesn't get confused?

Reply:
Yep, added in 507cc5319bdb4adbd706ad866cd591d087de9dfc